src/bios/mod.rs

```diff
@@ -32,6 +34,12 @@ impl BiosBoot {
         self
     }
 
+    /// Add a JSON configuration file to the disk image
+    pub fn set_config_file(&mut self, config_file_path: &Path) -> &mut Self {
```

Comment:
I would prefer to take the `BootloaderConfigFile` struct as argument here. This way, we don't need to write the JSON files by hand and we can statically ensure that the deserialization works.